src/client.js

```diff
@@ -128,20 +137,110 @@ class Client extends events.EventEmitter {
     }
 
     if (this.running) {
-      const newConn = new Connection(this, ++this.connectionCounter);
+      const newConn = new Connection(this);
+
+      // don't actually start connecting until at lesat minReconnectionInterval has passed
+      const earliestConnectionTime = this.lastConnectionTime + this._minReconnectionInterval;
+      const now = new Date().getTime();
+      setTimeout(() => {
+        this.lastConnectionTime = new Date().getTime();
+        newConn.connect();
+      }, now < earliestConnectionTime ? earliestConnectionTime - now : 0);
+
       newConn.once('connected', () => {
         this.emit('connected', newConn);
       });
       newConn.once('finished', () => {
-        this.connections = this.connections.filter(conn => conn.id !== newConn.id);
+        this.connections = this.connections.filter(conn => conn !== newConn);
       });
       this.connections.unshift(newConn);
     }
   }
+
+  /**
+   * Get a full object name, following the Pulse security model,
+   * `<kind>/<namespace>/<name>`.  This is useful for manipulating these objects
+   * directly, for example to modify the bindings of an active queue.
+   */
+  fullObjectName(kind, name) {
```

Comment:
Should we assert `kind` and `name` here to avoid creating something like `undefined/foo/undefined` or whatever? It probably isn't a big issue.